x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/use_results_rollup/index.tsx

```diff
@@ -71,49 +75,67 @@ const useStoredPatternRollups = (patterns: string[]) => {
 
     let ignore = false;
     const abortController = new AbortController();
-    const fetchStoredRollups = async () => {
-      const results = await getResults({ httpFetch, abortController, patterns, toasts });
-      if (results?.length && !ignore) {
-        setStoredRollups(Object.fromEntries(results.map(({ rollup }) => [rollup.pattern, rollup])));
+    const fetchStoredPatternResults = async () => {
+      const requests = patterns.map((pattern) =>
+        getStorageResults({ pattern, httpFetch, abortController, toasts }).then((results = []) => ({
+          pattern,
+          results: Object.fromEntries(
+            results.map((storageResult) => [
+              storageResult.indexName,
+              formatResultFromStorage({ storageResult, pattern }),
+            ])
+          ),
+        }))
+      );
+      const patternResults = await Promise.all(requests);
+      if (patternResults?.length && !ignore) {
+        setStoredPatternResults(patternResults);
       }
     };
 
-    fetchStoredRollups();
+    fetchStoredPatternResults();
     return () => {
       ignore = true;
     };
   }, [httpFetch, patterns, toasts]);
 
-  return storedRollups;
+  return storedPatternResults;
 };
 
 export const useResultsRollup = ({ ilmPhases, patterns }: Props): UseResultsRollup => {
   const { httpFetch, toasts } = useDataQualityContext();
   const [patternIndexNames, setPatternIndexNames] = useState<Record<string, string[]>>({});
   const [patternRollups, setPatternRollups] = useState<Record<string, PatternRollup>>({});
 
-  const storedPatternsRollups = useStoredPatternRollups(patterns);
+  const storedPatternsResults = useStoredPatternResults(patterns);
 
   useEffect(() => {
-    if (!isEmpty(storedPatternsRollups)) {
-      setPatternRollups((current) => ({ ...current, ...storedPatternsRollups }));
+    if (!isEmpty(storedPatternsResults)) {
+      setPatternRollups((current) => {
+        const updatedRollups = { ...current };
+        storedPatternsResults.forEach((storedPatternResult) => {
+          const { pattern, results } = storedPatternResult;
+          updatedRollups[pattern] = {
+            ...current[pattern],
+            pattern,
+            results,
+          };
+        });
+        return updatedRollups;
```

Comment:
Consider something like:

```ts
        const updatedRollups = storedPatternsResults.reduce((acc, storedPatternResult) => {
          const { pattern, results } = storedPatternResult;

          return {
            ...acc,
            [pattern]: {
              ...current[pattern],
              pattern,
              results,
            },
          };
        }, current);

        return updatedRollups;
      });
```

as an alternative to the local mutation of `const updatedRollups`

Reply:
This approach is less efficient since it does a copy of the entire structure on every iteration, I unconsciously tend to avoid doing this. However, since the number of patterns in the data view can not impact negatively, I think we are good with the _reduce_ approach. It looks more straightforward.